src/Domain/DomainHelpers.hpp

```diff
@@ -13,6 +13,8 @@
 #include "Domain/BlockNeighbor.hpp"
 #include "Utilities/ConstantExpressions.hpp"
 
+struct pair_of_sets;
```

Comment:
* To me this name is strongly suggestive of a type based on `std::pair<std::set<A>, std::set<B>>`, which this is not. True, the data placed in the struct _should_ be set-like... but it is not the job of the struct's _name_ to enforce this. If we do want to enforce it, the type should be changed. Otherwise, could this be called something like `VectorPair` (if we prefer to describe what the struct _is_) or`PairOfIndentifiedPoints` / `PairOfConformingPoints` (if we prefer to describe what the struct _is for_)?
* Types are CamelCase